desktop/portal/document.go

```diff
@@ -0,0 +1,82 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package portal
+
+import (
+	"fmt"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strings"
+
+	"github.com/snapcore/snapd/dbusutil"
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/i18n"
+)
+
+const (
+	documentPortalBusName    = "org.freedesktop.portal.Documents"
+	documentPortalObjectPath = "/org/freedesktop/portal/documents"
+	documentPortalIface      = "org.freedesktop.portal.Documents"
+)
+
+var (
+	userCurrent     = user.Current
+	osGetenv        = os.Getenv
+)
+
+type Document struct {
+	xdgRuntimeDir string
+}
+
+func (p *Document) GetUserXdgRuntimeDir() (string, error) {
+	if p.xdgRuntimeDir == "" {
+		u, err := userCurrent()
+		if err != nil {
+			return "", fmt.Errorf(i18n.G("cannot get the current user: %s"), err)
+		}
+		p.xdgRuntimeDir = filepath.Join(dirs.XdgRuntimeDirBase, u.Uid)
+	}
+	return p.xdgRuntimeDir, nil
+}
+
+func (p *Document) GetDefaultMountPoint() (string, error) {
+	xdgRuntimeDir, err := p.GetUserXdgRuntimeDir()
+	if err != nil {
+		return "", err
+	}
+
+	return filepath.Join(xdgRuntimeDir, "doc"), nil
+}
```

Comment:
The second yes, it should be kept here because it belongs to this interface. The first one would be better placed in some generic type (ideally a generic one for the Desktop portal), but moving it is not trivial if we want to cache its return value. I've added a comment about it, I suggest leaving it here and eventually come back to it if we see a need to reuse it.